sdk/evaluation-context.json

```diff
@@ -53,12 +123,121 @@
+                    "title": "Rules",
+                    "type": "array"
+                }
+            },
+            "required": [
+                "key",
+                "name",
+                "rules"
+            ],
+            "title": "SegmentContext",
+            "type": "object"
+        },
+        "SegmentCondition": {
+            "description": "Represents a condition within a segment rule for feature flag evaluation.",
+            "properties": {
+                "path": {
+                    "description": "The path to the trait or value in the evaluation context.",
+                    "title": "Property",
+                    "type": "string"
+                },
+                "operator": {
+                    "description": "The operator to use for evaluating the condition.",
+                    "title": "Operator",
+                    "type": "string",
+                    "enum": [
+                        "EQUAL",
+                        "GREATER_THAN",
+                        "LESS_THAN",
+                        "LESS_THAN_INCLUSIVE",
+                        "CONTAINS",
+                        "GREATER_THAN_INCLUSIVE",
+                        "NOT_CONTAINS",
+                        "NOT_EQUAL",
+                        "REGEX",
+                        "PERCENTAGE_SPLIT",
+                        "MODULO",
+                        "IS_SET",
+                        "IS_NOT_SET",
+                        "IN"
+                    ]
+                },
+                "value": {
+                    "description": "The value to compare against the trait or context value.",
+                    "title": "Value",
+                    "type": "string"
+                }
+            },
+            "required": [
+                "path",
+                "operator",
+                "value"
+            ],
+            "title": "SegmentCondition",
+            "type": "object"
+        },
+        "SegmentRule": {
+            "description": "Represents a rule within a segment for feature flag evaluation.",
+            "properties": {
+                "type": {
+                    "description": "Segment rule type. Represents a logical quantifier for the conditions and sub-rules.",
+                    "title": "Type",
+                    "type": "string",
+                    "enum": [
+                        "ALL",
+                        "ANY",
+                        "NONE"
+                    ]
+                },
+                "conditions": {
+                    "description": "Conditions that must be met for the rule to apply.",
+                    "items": {
+                        "$ref": "#/$defs/SegmentCondition"
+                    },
+                    "title": "Conditions",
+                    "type": "array"
+                },
+                "sub_rules": {
+                    "items": {
+                        "$ref": "#/$defs/SegmentRule"
+                    },
```

Comment:
Do we have to, or can we, ensure there's a depth limit to recursion of this field?

Reply:
This was also recursive in the SegmentSerializer until recently. In order to prevent it, here's the [strategy I took](https://github.com/Flagsmith/flagsmith/blob/4018f07d423fffde133fd6512e06c52f7ade2e37/api/segments/serializers.py#L47-L88).

I agree there is little gain here as the serializer now refuses more than one level of rule nesting, so the improvement is optional.